src/main/java/bookbob/Main.java

```diff
@@ -78,7 +78,23 @@ public static void main(String[] args) throws IOException {
                         break;
                     }
 
-                    String visitDateString = input.substring(visitStart + 2).trim();
+                    if (visitStart == -1) {
+                        System.out.println("Please provide the visit date.");
+                        logger.log(Level.INFO, "Visit date is not provided");
+                        break;
+                    }
+
+                    String[] fieldIndicators = { "p/", "ha/", "dob/", "al/", "s/", "mh/" };
+                    int minIndex = input.length();
+
+                    for (String fieldIndicator : fieldIndicators) {
```

Comment:
this is to locate all the possible prefixes that might appear after the visit date prefix (v/) from the the user input string 
(since the other prefixes are optional fields)